seismicpro/src/gather.py

```diff
@@ -961,15 +974,139 @@ def plot(self, figsize=(10, 7), **kwargs):
         self : Gather
             Gather unchanged.
         """
-        vmin, vmax = self.get_quantile([0.1, 0.9])
-        default_kwargs = {
-            'cmap': 'gray',
-            'vmin': vmin,
-            'vmax': vmax,
-            'aspect': 'auto',
+        # Make the axis divisible to further plot colorbar and header subplot
+        divider = make_axes_locatable(ax)
+
+        # Plot the gather depending on the mode passed
+        plotters_dict = {
+            "seismogram": partial(self._plot_seismogram, divider=divider),
+            "wiggle": self._plot_wiggle,
         }
-        default_kwargs.update(kwargs)
-        plt.figure(figsize=figsize)
-        plt.imshow(self.data.T, **default_kwargs)
-        plt.show()
+        if mode not in plotters_dict:
+            raise ValueError(f"Unknown mode {mode}")
+        plotters_dict[mode](ax, **kwargs)
+
+        # Add headers scatter plot if needed
+        if event_headers is not None:
+            self._plot_headers(ax, event_headers)
+
+        # Add a top subplot for given header if needed and set plot title
+        top_ax = ax
+        if header_on_top is not None:
+            top_ax = self._plot_top_subplot(ax=ax, divider=divider, header_values=self[header_on_top])
+        top_ax.set_title(**as_dict(title, key='label'))
+
+        # Set axis ticks
+        if x_ticker is None:
+            x_ticker = self.sort_by if self.sort_by is not None else "index"
+        self._set_ticks(ax, axis="x", ticker=x_ticker)
+        self._set_ticks(ax, axis="y", ticker=y_ticker)
         return self
+
+    def _plot_seismogram(self, ax, divider, colorbar=False, qvmin=0.1, qvmax=0.9,**kwargs):
+        vmin, vmax = self.get_quantile([qvmin, qvmax])
+        kwargs = {"cmap": "gray", "aspect": "auto", "vmin": vmin, "vmax": vmax, **kwargs}
+        img = ax.imshow(self.data.T, **kwargs)
+        if not isinstance(colorbar, (bool, dict)):
+            raise ValueError(f"colorbar must be bool or dict but {type(colorbar)} was passed")
+        if colorbar is not False:
+            colorbar = {} if colorbar is True else colorbar
+            cax = divider.append_axes("right", size="5%", pad=0.05)
+            ax.figure.colorbar(img, cax=cax, **colorbar)
+
+    def _plot_wiggle(self, ax, std=0.5, color=None):
+        color = ["black"] if color is None else to_list(color)
+        if len(color) == 1:
+            color = color * self.n_traces
+        elif len(color) != self.n_traces:
+            raise ValueError('The number of items in `color` must match the number of plotted traces')
+
+        y_coords = np.arange(self.n_samples)
+        traces = np.arange(self.n_traces).reshape(-1, 1) + std * self.data / np.std(self.data)
```

Comment:
if self.data contains zero traces, we'll have division by zero
```suggestion
        traces = np.arange(self.n_traces).reshape(-1, 1) + std * self.data / (np.std(self.data) + 1e-10)
```

Reply:
Done